apps/condo/domains/meter/tasks/sendSubmitMeterReadingsPushNotifications.js

```diff
@@ -112,40 +146,62 @@ const sendSubmitMeterReadingsPushNotifications = async () => {
             offset: state.offset,
             pageSize: state.pageSize,
         })
-        const meterReadings = await readMeterReadings({ context, meters })
         const organizations = await readOrganizations({ context, meters })
+        const reportingPeriods = await readMeterReportingPeriods({ context, organizations })
+        const meterReadings = await readMeterReadings({ context, meters, reportingPeriods })
+
+        const metersWithoutReadings = []
+        const periodsByProperty = []
+        const periodsByOrganization = []
+        let defaultPeriod = null
+
+        for (let period of reportingPeriods) {
+            if (isNil(period.organization) && isNil(period.property)) defaultPeriod = period
+            else if (!isNil(period.organization) && isNil(period.property)) periodsByOrganization.push(period)
+            else periodsByProperty.push(period)
+        }
 
-        // right join data to metersPage
-        const metersWithoutReadings = rightJoin(
-            meters,
-            meterReadings,
-            (meter, item) => meter.id === item.meter.id,
-            (meter, readings) => ({ meter, readings })
-        )
-            .filter(({ readings }) => readings.length === 0)
-            .map(({ meter }) => meter)
+        for (const meter of meters) {
+            const period = periodsByProperty.find(({ property }) => property.id === meter.property.id) ??
+                periodsByOrganization.find(({ organization }) => organization.id === meter.organization.id) ??
+                defaultPeriod
+
+            if (isNil(period)) continue
+
+            const notifyStartDay = get(period, 'notifyStartDay')
+            const notifyEndDay = get(period, 'notifyEndDay')
+
+            const readingsOfCurrentMeter = meterReadings.filter(reading => (
+                reading.meter.id === meter.id &&
+                checkIsDateStartOrEndOfPeriod(reading.date, state.startTime, notifyStartDay, notifyEndDay)
+            ))
+
+            const isTodayStartOrEndOfPeriod = checkIsDateStartOrEndOfPeriod(state.startTime, state.startTime, notifyStartDay, notifyEndDay)
+            const isEndPeriodNotification = dayjs(state.startTime).format('YYYY-MM-DD') === dayjs(state.startTime.format('YYYY-MM-DD')).set('date', notifyEndDay).format('YYYY-MM-DD')
```

Comment:
I guess you can use it in this way: dayjs(state.startTime.format('YYYY-MM-DD')) -> dayjs(state.startTime)